### PR TITLE
Generalize ApprovalDenied/RejectionDenied to ParticipationDenied

### PR DESCRIPTION
**Pull request by matthias-geuchen**
### Reason for this PR:
- to avoid leaking sensitive data, transactions should only be executable by users required to approve the transaction

### Changes in this PR:
- generalize approval denied and rejection denied errors to participation denied error
- add participation denied error to errors throwable by transactions utilizing our operation workflow
- outsource errors related to our operation workflow from transactions to general error definition

### CODE REVIEW

**Review by NikoBergemann — APPROVED** [review]

File: hlf/chaincode/contracts/operation.md [hunk 1]
```diff
@@ -31,16 +31,16 @@ The Errors returned are defined [here](../errors.md#Errors).
     - [GenericError](../errors.md#GenericError) 
       ```json
       {
-        "type": "HLApprovalDenied",
-        "title": "You are not allowed to approve the given operation"
+        "type": "HLParticipationDenied",
+        "title": "You are not allowed to participate in the given operation"
```
Comment (dhelmert) [inline hlf/chaincode/contracts/operation.md]:
```suggestion
        "title": "You are not allowed to approve or reject the given operation"
```

File: hlf/chaincode/contracts/operation.md [hunk 2]
```diff
@@ -74,16 +74,16 @@ The Errors returned are defined [here](../errors.md#Errors).
...
```
Comment (dhelmert) [inline hlf/chaincode/contracts/operation.md]:
```suggestion
        "title": "You are not allowed to approve or reject the given operation"
```

File: hlf/chaincode/contracts/operation.md [hunk 3]
```diff
@@ -123,19 +123,19 @@ The Errors returned are defined [here](../errors.md#Errors).
...
       -  Description: Success, returns the list of operations.
 
-    - [GenericError](errors.md#GenericError) 
+    - [GenericError](../errors.md#GenericError) 
       ```json
       {
-        "type": "HLRejectionDenied",
-        "title": "You are not allowed to reject the given operation"
+        "type": "HLParticipationDenied",
+        "title": "You are not allowed to participate in the given operation"
```
Comment (dhelmert) [inline hlf/chaincode/contracts/operation.md]:
```suggestion
        "title": "You are not allowed to approve or reject the given operation"
```

**Review by dhelmert — CHANGES_REQUESTED** [review]
While i approve of generalizing the approval and rejection errors into Participation errors, I do not think that putting the "HLInsufficientApprovals" and "ExecutionImpossible" errors into the errors file is a good idea.
A problem with the current api is that whenever you inspect a specific call, there can be some really important information (like which initiate an operation or the fact that everything can be a proposal) somewhere else. I think that this is, in the example cases, acceptable. 
However, it is my opinion that having the errors a call can return in a general file, instead of at the specific call, significantly harms readability.

File: hlf/chaincode/errors.md [hunk 4]
```diff
@@ -36,3 +36,35 @@ name:
...
+    "title": "The operation is not in pending state"
+  }
+  ```
+  - Description: This error is returned, if the operation for this transaction is not pending.
+
+- [GenericError](#GenericError) 
+  ```json
+  {
+    "type": "HLParticipationDenied",
+    "title": "You are not allowed to participate in the given operation"
```
Comment (matthias-geuchen) [inline hlf/chaincode/errors.md]:
```suggestion
    "type": "HLAccessDenied",
    "title": "You are not allowed to execute in the given transaction"
```

File: hlf/chaincode/errors.md [hunk 5]
```diff
@@ -36,3 +36,35 @@ name:
 
 reason:
 - a textual representation of why the parameter is invalid
+
+
+# Specific Errors
+
+In the following, the specific error that can be returned by every transaction not part of the [OperationContract](./contracts/operation.md#OperationApi)  due to our operation workflow.
```
Comment (matthias-geuchen) [inline hlf/chaincode/errors.md]:
```suggestion
In the following, the specific error that can be returned by every transaction due to our operation workflow.

```

File: hlf/chaincode/errors.md [hunk 6]
```diff
@@ -36,3 +36,43 @@ name:
...
+
+
+## <a id="AllOperations" />All Operations
+
+In the following, the specific errors that can additionally be returned by every operation are defined.
+
+- [GenericError](#GenericError) 
+  ```json
+  {
+    "type": "HLParticipationImpossible",
```
Comment (RBeierling) [inline hlf/chaincode/errors.md]:
```suggestion
    "type": "HLExecutionImpossible",
```

**Review by dhelmert — APPROVED** [review]

**Review by RBeierling — APPROVED** [review]

**Review by bastihav — APPROVED** [review]

**Pull request merged**